### ISSUE TITLE
Find missing parameters in smirff99Frosst for DrugBank

### ISSUE THREAD

**Issue by bannanc** [open-forcefield-group/smarty#216]
Since the ipython notebook for converting Frcmod files to smirff ffxml files is in this repository, I'm cross listing issue#14 from smirff99Frosst https://github.com/open-forcefield-group/smirff99Frosst/issues/14

**Comment by bannanc** [comment]
This issue is related to #226 but I thought enumerating the generic parameters here made more sense since #226 is about the molecule set and this issue is about smirff99Frosst

I apologize for the length, but I'm yet to find a concise way of writing out my thoughts on missing parameters. 

Based on the current MiniDrugBank set, the following types of fragments are still assigned generic parameters with smirff99Frosst. I have gotten very used to looking at SMIRKS patterns so I tried to write some that are specific enough to catch the missing chemistry. I will include some images of particularly difficult functional groups. 
Also since they're atomtyped I printed the atomtypes for each fragment which might make looking up parameters easier (though they may be missing from the parm@frosst force field too). 

## Molecules/Fragments that caused major problems

### N-P=S 

This molecule had a total of 18 generic torsions around the nitrogen-phosphorous=sulfur group
![2fa1ada2-4061-4986-8884-845b176eca2d](https://cloud.githubusercontent.com/assets/11968541/23964924/74c2d0c4-0973-11e7-914f-ad21bca93628.png)

### Peroxides

We're missing torsions for a normal C-O-O-C peroxide, but there was a molecule with a C-N-O-O-H group and another with a H-O-O-O-C group!
 
![f48a7c01-7901-4856-8209-ab993b426c66](https://cloud.githubusercontent.com/assets/11968541/23964995/be5f9578-0973-11e7-92dd-262aeb7a2bc3.png)
 
![d2af3e49-f350-4718-976c-e791fdaed524](https://cloud.githubusercontent.com/assets/11968541/23964998/c2e260ee-0973-11e7-9085-3a8d652bf2c5.png)

### Trivalent, negative, sulfur

This group came up twice in the MiniDrugBank set, and even more in the full DrugBank database

![94983129-872d-4cf8-923a-6fc52b001a56](https://cloud.githubusercontent.com/assets/11968541/23965007/cad9eec0-0973-11e7-934e-f720d25c8526.png)

### phosphate group with nitrogen

This is a section of molecule MiniDrugBank\_167

![38b90754-890b-4fef-a45c-df1f66f24c62](https://cloud.githubusercontent.com/assets/11968541/23965096/1c81b1b8-0974-11e7-9c63-1d690cafc76c.png)

### Weird nitrile group

Oddly this molecule did not come up in the torsions... so the torsion around C#N must get grouped with a generic `*~C~N~*` torsion or something...

![1bc1e75c-a096-410c-bd09-a6a6a08e149d](https://cloud.githubusercontent.com/assets/11968541/23965310/cc6c3cf6-0974-11e7-92d1-6f0c4032545e.png)

## Vdw

None, which is good, there is a generic non-bond in the full DrugBank (for Helium...)

## Bonds

There were 7 molecules with a combination of what looks like 4 missing bonds

* `[#15:1]-[#16-1:2]`
     - That is phosphorous singly bound to a sulfur with a negative charge
     - P~Su
* `[#16-1:1]=[#8:2]`
     - SO~O2
* `[#6:1]#[#7+1X2:2]`
     - This is a nitrile group with 4 bonds to the nitrogen, see above
     - NL~C2
* C=S
     - like a carbonyl group, but sulfur, it is definitely in parm@frosst
     - CM~SD

## Angles

Only one generic angle, for the weird nitrile group

* `[#6:1]#[#7+1X2:2]-[#6:3]`
     - CT\~NL\~C2

## Torsions

There are 21 molecules with generic torsions, it looks like maybe 14 different torsions? I'll try to group them below by center bond. It is possible they could be more generic than the ones I'm writing out. I also have my description of the torsion, then at least one example set of atom types in the sub bullet points

#### Phosphorous-Nitrogen

Basically all of them in MiniDrugBank\_180 above 

* `S=P-N-C`
    - SD\~P\~N3\~CJ
* `N-P-N-C`
    - N3\~P\~N3\~CJ
* `C-N-P=O`
    - CT\~N3\~P\~O
    - CR\~N2\~P\~O
* `C-N-P-O`
    - CR\~N2\~P\~OH
* `C-N-P-C`
    - CT\~N3\~P\~CT
* `H-N-P-O`
    - H\~N2\~P\~OH
* `H-N-P=O`
    - H\~N2\~P\~O

#### Peroxides (and similar)

These might all need to fall under a `[*:1]~[#8:2]~[#8:3]~[*:4]` type torsion, I doubt the parm@frosst forcefield could have actually handed a H-O-O-O group or a H-O-O-N group

* `C-O-O-C`
    - C\~OS\~OS\~C
    - CT\~OS\~OS\~CT
* `H-O-O-C`
    - HO\~OH\~OS\~CT
* `H-O-O-N`
    - HO\~OH\~OS\~N*
* `H-O-O-O`
    - HO\~OH\~OS\~OS
* `C-O-O-O`
    - CA\~OS\~OS\~OH

#### Trivalent negative sulfur 

There are more than one molecule with the sulfite like group in MiniDrugBank\_232 

* `[#8:1]=[#16X3-1:2]-[#6:3]-[#1:4]`
    - O2\~SO\~CT\~H1
    - I would not expect that hydrogen to be an H1, the carbon is not attached to an electron withdrawing group...
* `[#8:1]=[#16X3-1:2]-[#6:3]-[#6:4]`
    - O2\~SO\~CT\~CT

**Comment by bannanc** [comment]
I'm going to try to list my proposed changes for each parameter type here. It should be noted that these are the ones I thought of as "easy" to fix. Anything that is missing in the parm@Frosst world are being made into their own issues in smirff99Frosst.

**BONDS**
- [x] Change `[#16X3+0,#16X4]` to `[#16X3,#16X4]`
    * These are in smirff99Frosst but with more decorators than what was in parm@frosst. Looking at xpat, the SO atom type was any tri- or tetra-valent sulfur. The corresponding bond (and so I assume possibly angle/torsion?) in smirff has the SMARTS `[#16X3+0,#16X4]`
- [x] Add C=S `[#6:1]=[#16:2]`
    * There are two parm@frosst and they agree with k=300.00 and r=1.7
```
CA-SD        300.000     1.7     calc B3LYP/cc-pVTZ  (estimated force constant)
CM-SD        300.000     1.688   b3lyp/6-31G(d) opt  July_17_2007
```
- [x] Remove 'X1' from nitrile bond
   * C#N bond in smirff has pattern `[#6:1]#[#7X1:2]` I think we should remove the 'X1' for now 
- [x] more generic C-S [#16:1]-[#6:2]
   * Right now the S-C bond is only for tetrahedral carbon, for the sake of minimal parameters and maximal chemical space, I think it makes more sense to remove the 'X4' from the carbon in that bond

**ANGLE**
- [x] Add angle centered on nitrile nitrogen `[*:1]~[#7X2:2]~[*:3]`
   * All the sp carbons and nitrile nitrogens in parm@frosst are the same with k=70 and theta = 180.0 with a note about being standard sp carbon or guessing in the case of nitrogen
   * add above this line: `[*:1]~[#7X2+0:2]~[*:3]   70.  120.    Frosst generic divalent conjugated Nsp2`

**Comment by davidlmobley** [comment]
Is this the full list, or something you're updating as you go?

**Comment by bannanc** [comment]
That is the list of Bonds and Angles, I'm still working on Torsions. Does it make more sense for me to just start a WIP pull request with everything?

**Comment by davidlmobley** [comment]
You can update the list above of proposed changes as you go, I think, and also resolve via a PR. You can open the pr whenever it makes sense to do so; I don't see a major reason to prefer opening it now or later, whichever makes the most sense in your workflow.

**Comment by bannanc** [comment]
Should be closed by #232
